### PR TITLE
Unified testing resources

### PR DESCRIPTION
**Pull request by RagdollGoDown**
# UNIFIED TESTING RESOURCES
## Description
- What has been changed?
Added testing resources, specifically a list of ingredients, recipes and users. 
https://github.com/PlateSwipe/PlateSwipe/pull/222/files#diff-0e283674ff6eb1d54f92f45648d7e98188b83ab556621d13bb948902fb922818
A few corrections to the `UserViewModelTest` file as it would compare lists of different types and usually not take into account that we weren't only testing empty lists

- Why was this change made?
This way, if we want to change one of this data classes, it will be a lot less work. I also got rid of the hardcoded values for a lot of the tests, reducing the need for maintenance.
## Checklist
- [x] All CI checks passed.

### CODE REVIEW

**Review by akshire — COMMENTED** [review]
Hey @RagdollGoDown, great PR! 🎉

This change is a fantastic improvement for our testing approach. By centralizing all test examples into a single file, you've given us a unified way of testing across all our classes. This will make it much easier to ensure consistency and efficiency when writing or updating tests.

The ability to test edge cases across multiple screens seamlessly is a huge win—yay for streamlined testing resources! 🚀

Thanks for this thoughtful and impactful contribution. Keep up the great work! 😊

File: app/src/androidTest/java/com/android/sample/e2e/End2EndTest.kt [hunk 1]
```diff
@@ -289,16 +273,18 @@ class EndToEndTest {
         .onNodeWithTag(RECIPE_FAVORITE_ICON_TEST_TAG, useUnmergedTree = true)
         .performClick()
 
+    // TODO fix
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/androidTest/java/com/android/sample/e2e/End2EndTest.kt:276]: Oh no sorry, that's a mistake on my part.

File: app/src/main/java/com/android/sample/ui/utils/SampleClasses.kt
[the remark this answers is not in the document]
Comment (akshire) [reply app/src/main/java/com/android/sample/ui/utils/SampleClasses.kt:1]: Ok ok, I understand, maybe we can later implement something that is called a `testFixture` folder that will enable us to access resoureces in the `test` folder and also in the `androidTest` folder:
https://stackoverflow.com/questions/64133013/gradle-test-fixtures-plugin-and-core-module-dependencies#64302254

But for the moment LGTM

**Review by akshire — APPROVED** [review]
LGTM! all suggestions were discussed.
